lib/replica_pools/query_cache.rb

```diff
@@ -18,29 +18,20 @@ module QueryCache
     # select_all is trickier. it needs to use the leader
     # connection for cache logic, but ultimately pass its query
     # through to whatever connection is current.
-    def select_all(*args, **kwargs)
-      relation, name, raw_binds = args
-
-      if !query_cache_enabled || locked?(relation)
-        return route_to(current, :select_all, *args, **kwargs)
+    def select_all(arel, name = nil, binds = [], *args, preparable: nil, **kwargs)
+      if !query_cache_enabled || locked?(arel)
```

Comment:
I moved `relation`, `name`, and `raw_binds` into the method signature as `arel`, `name`, and `binds` because I thought it made the method much easier to understand / read, and also easier to compare with the method we are imitating from activerecord.